### PR TITLE
Add Trino 459 release notes

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Assemble the release notes for the upcoming Trino 459 release.

Also see https://github.com/trinodb/trino/pulls?q=is%3Apr+is%3Aclosed+milestone%3A459

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 17 Sep 2024

* #23475 ✅ rn ✅ docs

## 18 Sep 2024

* #23476 ✅ rn ✅ docs
* #23478 ✅ rn ✅ docs
* #22932 ✅ rn ✅ docs
* #23231 ✅ rn ✅ docs
* #23482 ✅ rn ✅ docs
* #23443 ✅ rn ✅ docs
* #23472 ✅ rn ✅ docs
* #23487 ✅ rn ✅ docs
* #23469 ✅ rn ✅ docs
* #23425 ✅ rn ✅ docs

## 19 Sep 2024

* #23499 ✅ rn ✅ docs
* #23494 ✅ rn ✅ docs
* #23504 ✅ rn ✅ docs
* #23506 ✅ rn ✅ docs
* #14859 ✅ rn ✅ docs
* #23509 ✅ rn ✅ docs
* #23486 ✅ rn ✅ docs

## 20 Sep 2024

* #23510 ✅ rn ✅ docs
* #23514 ✅ rn ✅ docs
* #23454 ✅ rn ✅ docs
* #23515 ✅ rn ✅ docs
* #23508 ✅ rn ✅ docs
* #23471 ✅ rn ✅ docs
* #23481 ✅ rn ✅ docs
* #23496 ✅ rn ✅ docs

## 21 Sep 2024

* #23516 ✅ rn ✅ docs
* #23519 ✅ rn ✅ docs
* #23521 ✅ rn ✅ docs

## 22 Sep 2024

* #23522 ✅ rn ✅ docs

## 23 Sep 2024

* #23529 ✅ rn ✅ docs
* #23408 ✅ rn ✅ docs
* #23526 ✅ rn ✅ docs
* #23520 ✅ rn ✅ docs
* #23457 ✅ rn ✅ docs
* #23534 ✅ rn ✅ docs
* #23524 ✅ rn ✅ docs
* #23535 ✅ rn ✅ docs
* #23536 ✅ rn ✅ docs

## 24 Sep 2024

* #23533 ✅ rn ✅ docs
* #23541 ✅ rn ✅ docs
* #23175 ✅ rn ✅ docs
* #23542 ✅ rn ✅ docs
* #23539 ✅ rn ✅ docs
* #23543 ✅ rn ✅ docs
* #23456 ✅ rn ✅ docs
* #23552 ✅ rn ✅ docs
* #23549 ✅ rn ✅ docs

## 24 Sep 2024

* #23546 ✅ rn ✅ docs
* #23556 ✅ rn ✅ docs
* #23547 ✅ rn ✅ docs
* #23560 ✅ rn ✅ docs
* #23568 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-459.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+## BigQuery connector
+
+* Add support for dereference pushdown. ({issue}`23443`)
+* Fix query failure when the `bigquery.service-cache-ttl` property isn't `0ms`
+  and case insensitive name matching is enabled. ({issue}`23481`)
+
+## ClickHouse connector
+
+* Unconditionally pushdown `varchar` predicates. ({issue}`23516`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-459.md:28]: Not sure if we want to reword this.

File: docs/src/main/sphinx/release/release-459.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+## CLI
+
+* Display data sizes and rates with binary (1024-based) abbreviations such as
+  `KiB` and `MiB`. Add flag `--decimal-data-size` to use decimal (1000-based)
+  values and abbreviations such as `KB` and `MB`. (#13054)
+
+## BigQuery connector
+
+* Add support for dereference pushdown. ({issue}`23443`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-459.md:22]: "dereference pushdown" is planner/IR jargon. Rephrase as "Improve performance of queries that access only a subset of fields from nested data".

File: docs/src/main/sphinx/release/release-459.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+* Fix query failure when the `bigquery.service-cache-ttl` property isn't `0ms`
+  and case insensitive name matching is enabled. ({issue}`23481`)
+
+## ClickHouse connector
+
+* Unconditionally pushdown `varchar` predicates. ({issue}`23516`)
+
+## Delta Lake connector
+
+* Allow configuring maximum HTTP requests in
```
Comment (martint) [inline docs/src/main/sphinx/release/release-459.md:32]: What does "maximum HTTP requests" mean? Is it referring to "concurrent HTTP requests", "connections", something else?

File: docs/src/main/sphinx/release/release-459.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
+# Release 459 (25 Sep 2024)
+
+## General
+
+* Improve error reporting on input validation for the `approx_percentile`
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-459.md:5]: Removed

File: docs/src/main/sphinx/release/release-459.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-459.md:28]: Improve performance for queries involving conditions with `varchar` data.

Comment (mosabua) [reply docs/src/main/sphinx/release/release-459.md:28]: maybe

File: docs/src/main/sphinx/release/release-459.md (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-459.md:32]: In the docs we added 

 Maximum [integer](prop-type-integer) number of concurrent HTTP requests to
    Azure on every node. Defaults to double the number of processors on the
    node. Minimum `1`. Use this property to reduce the number of connections
    when you observe too many connections from your Trino cluster nodes to
    Azure Storage and encounter rate limiting issues.

I adjusted a bit towards that..

File: docs/src/main/sphinx/release/release-459.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+* Allow configuring maximum HTTP requests in
+  [](/object-storage/file-system-azure) with `azure.max-http-requests`.
+  ({issue}`22915`)
+* Allow disabling caching of Delta Lake transaction logs when filesystem caching
+  is enabled with the `delta.fs.cache.disable-transaction-log-caching` property. ({issue}`21451`)
+* Add support for WASB to [](/object-storage/file-system-azure). ({issue}`23511`)
+* Improve cache hit ratio of the [](/object-storage/file-system-cache). ({issue}`23172`)
+* Fix incorrect results when writing deletion vectors. ({issue}`23229`)
+* Fix failures for queries with distinct aggregations on metadata tables. ({issue}`23529`)
```
[the remark this answers is not in the document]
Comment (mosabua) [reply docs/src/main/sphinx/release/release-459.md:40]: update ok?

File: docs/src/main/sphinx/release/release-459.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+* Allow configuring maximum concurrent HTTP requests to Azure on every node in
+  [](/object-storage/file-system-azure) with `azure.max-http-requests`.
+  ({issue}`22915`)
+* Allow disabling caching of Delta Lake transaction logs when filesystem caching
+  is enabled with the `delta.fs.cache.disable-transaction-log-caching` property. ({issue}`21451`)
+* Add support for WASB to [](/object-storage/file-system-azure). ({issue}`23511`)
+* Improve cache hit ratio for the [](/object-storage/file-system-cache). ({issue}`23172`)
+* Fix incorrect results when writing [deletion
+  vectors](https://docs.delta.io/latest/delta-deletion-vectors.html). ({issue}`23229`)
+* Fix failures for queries with containing aggregations with a `DISTINCT`
```
Comment (martint) [inline docs/src/main/sphinx/release/release-459.md:40]: This needs to be applied to other connectors below.

File: docs/src/main/sphinx/release/release-459.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  clause on metadata tables. ({issue}`23529`)
+
+## Elasticsearch connector
+
+* Fix failures for `count(*)` queries with predicates containing non-ASCII
```
Comment (martint) [inline docs/src/main/sphinx/release/release-459.md:45]: This needs to be applied to opensearch, too

**Pull request merged**